dev/cuda/gelu_forward.cu

```diff
@@ -44,6 +48,25 @@ __global__ void gelu_kernel(float* out, const float* inp, int N) {
     }
 }
 
+// Optimized GELU kernel using float4
+__global__ void gelu_kernel2(float4* out, const float4* inp, int N) {
+    int idx = (blockIdx.x * blockDim.x + threadIdx.x); 
+    int i = idx * 4; // Each thread handles 4 floats
+    if (i < N) {
```

Comment:
if we assert N % 4 we can remove both if guards, which are annoying and ugly and create identation.